src/os/bluestore/BlueStore.cc

```diff
@@ -7732,8 +7729,10 @@ void BlueStore::_txc_state_proc(TransContext *txc)
 	  kv_queue_unsubmitted.push_back(txc);
 	  ++txc->osr->kv_committing_serially;
 	}
-	if (txc->had_ios)
+	if (txc->had_ios) {
 	  kv_ios++;
+	  ++txc->osr->txc_with_unstable_io;
```

Comment:
You are going to move this atomic op within the scope of the (big) kv_lock, which I think will slightly increase the overhead.